torchao/prototype/quantized_training/README.md

```diff
@@ -0,0 +1,42 @@
+# Quantized training
+
+This folder contains experimental work on quantized training (QT). The main difference from quantization-aware training (QAT) is that in QT, we don't keep a high-precision copy of model weights. We take inspirations from:
+- Q-GaLore: [[paper](https://arxiv.org/abs/2407.08296)] [[code](https://github.com/VITA-Group/Q-GaLore)]
+- AQT: [[related paper](https://arxiv.org/abs/2105.03536)] [[code](https://github.com/google/aqt)]
+
+Typically, low-precision weights cannot be trained directly due to quantization error: a small change in the quantized weight will be round down to zero. To tackle this problem, we use **stochastic rounding** for weight update. In simple terms, stochastic rounding will round up or down randomly, but with a higher chance if it is closer to that direction. For example, 0.8 will have 80% chance of rounding up and 20% of rounding down. It also follows that on average, stochastic rounding will estimate the floating point value exactly.
+
+There are 2 main benefits for training in this way:
+1. Reduce memory footprint. Also reduce communication bandwidth in distributed setting.
+2. What you train is what you serve (WYTIWYS).
```

Comment:
curious where you got the terminology from?

Reply:
From Google https://github.com/google/aqt?tab=readme-ov-file#features. First used in their first AQT paper I think (https://arxiv.org/pdf/2105.03536, section 3.3, pointer 5). The Character.AI also alludes to this, though not using the exact words. Actually QAT can be considered "What you train is what you serve too", though Google might not think so 😄.